pkg/model/iam.go

```diff
@@ -64,33 +68,63 @@ func (b *IAMModelBuilder) Build(c *fi.ModelBuilderContext) error {
 	// Generate IAM objects etc for each role
 	for _, role := range roles {
 		name := b.IAMName(role)
+		roleAsString := reflect.ValueOf(role).String()
 
 		var iamRole *awstasks.IAMRole
-		{
-			rolePolicy, err := b.buildAWSIAMRolePolicy()
-			if err != nil {
-				return err
+
+		arn := ""
+
+		if b.Cluster.Spec.CustomPolicies != nil && featureflag.CustomPoliciesSupport.Enabled() {
+			r := *(b.Cluster.Spec.CustomPolicies)
+
+			// The map is keyed like AdditionalPolicies, master / node
+			if a, ok := r[roleAsString]; ok {
+				glog.Warningf("Custom Policy Support is enabled, kops will use %s, for %s role, this is an advanced feature please use with great care", arn, roleAsString)
+				glog.Warningf("With great power comes great responsibility")
```

Comment:
I've been playing with `Lifecycle` stuff, so that we would validate the role but not create it.  I think we need that, rather then just warnings...